barretenberg/cpp/src/barretenberg/vm/avm/tests/mem_opcodes.test.cpp

```diff
@@ -221,21 +221,33 @@ TEST_F(AvmMemOpcodeTests, uninitializedValueMov)
     trace_builder.op_return(0, 0, 0);
     trace = trace_builder.finalize();
 
-    validate_mov_trace(false, 0, 0, 1, AvmMemoryTag::U0);
+    validate_mov_trace(false, 0, 0, 1, AvmMemoryTag::FF);
 }
 
 TEST_F(AvmMemOpcodeTests, indUninitializedValueMov)
 {
     // TODO(#9131): Re-enable once we have error handling on wrong address resolution
     GTEST_SKIP();
 
+    trace_builder.op_set(0, 1, 3, AvmMemoryTag::U32);
+    trace_builder.op_set(0, 4, 1, AvmMemoryTag::U32);
+    trace_builder.op_set(0, 5, 2, AvmMemoryTag::U32);
+    trace_builder.op_mov(3, 2, 3);
+    trace_builder.op_return(0, 0, 0);
+    trace = trace_builder.finalize();
+
+    validate_mov_trace(true, 0, 2, 3, AvmMemoryTag::FF, 5, 1, false);
+}
+
+TEST_F(AvmMemOpcodeTests, indUninitializedAddrMov)
+{
     trace_builder.op_set(0, 1, 3, AvmMemoryTag::U32);
     trace_builder.op_set(0, 4, 1, AvmMemoryTag::U32);
     trace_builder.op_mov(3, 2, 3);
     trace_builder.op_return(0, 0, 0);
     trace = trace_builder.finalize();
 
-    validate_mov_trace(true, 0, 2, 3, AvmMemoryTag::U0, 0, 1, true);
+    validate_mov_trace(true, 0, 2, 3, AvmMemoryTag::FF, 0, 1, true);
 }
 
```

Comment:
New test that ensures `tag_err` occurs if trying to read uninitialized memory as an address.